src/downloadextractthread.cpp

```diff
@@ -145,17 +100,48 @@ DownloadExtractThread::~DownloadExtractThread()
     qFreeAligned(_abuf[1]);
 }
 
+void DownloadExtractThread::_emitProgressUpdate()
+{
+    static QElapsedTimer timer;
+    bool firstProgressUpdate = false;
+    if (!_progressStarted) {
+        _progressStarted = true;
+        firstProgressUpdate = true;
+        timer.start();
+        qDebug() << "Started progress updates after successful drive opening";
+    }
+    
+    // Only emit progress updates every 100ms to avoid flooding (but always emit the first one)
```

Comment:
"at most every 100ms" ? (i.e. the updates may be less frequent for a slow SD card?)